src/index.ts

```diff
@@ -107,77 +107,78 @@ const useDimensions = <T extends HTMLElement | null>({
-            console.warn(borderBoxWarn);
-            warnedRef.current = true;
+        window.requestAnimationFrame(() => {
+          const { contentBoxSize, borderBoxSize, contentRect } = entry;
+
+          let boxSize = contentBoxSize;
+          if (useBorderBoxSize)
+            if (borderBoxSize) {
+              boxSize = borderBoxSize;
+            } else if (!warnedRef.current) {
+              console.warn(borderBoxWarn);
+              warnedRef.current = true;
+            }
+          // @juggle/resize-observer polyfill has different data structure
+          boxSize = Array.isArray(boxSize) ? boxSize[0] : boxSize;
+
+          const width = boxSize ? boxSize.inlineSize : contentRect.width;
+          const height = boxSize ? boxSize.blockSize : contentRect.height;
+
+          if (
+            width === prevSizeRef.current.width &&
+            height === prevSizeRef.current.height
+          )
+            return;
+
+          prevSizeRef.current = { width, height };
+
+          const e = {
+            currentBreakpoint: "",
+            width,
+            height,
+            entry,
+            observe,
+            unobserve,
+          };
+
+          if (breakpoints) {
+            e.currentBreakpoint = getCurrentBreakpoint(breakpoints, width);
+
+            if (e.currentBreakpoint !== prevBreakpointRef.current) {
+              if (onResizeRef.current) onResizeRef.current(e);
+              prevBreakpointRef.current = e.currentBreakpoint;
+            }
+          } else if (onResizeRef.current) {
+            onResizeRef.current(e);
           }
-        // @juggle/resize-observer polyfill has different data structure
-        boxSize = Array.isArray(boxSize) ? boxSize[0] : boxSize;
-
-        const width = boxSize ? boxSize.inlineSize : contentRect.width;
-        const height = boxSize ? boxSize.blockSize : contentRect.height;
-
-        if (
-          width === prevSizeRef.current.width &&
-          height === prevSizeRef.current.height
-        )
-          return;
-
-        prevSizeRef.current = { width, height };
-
-        const e = {
-          currentBreakpoint: "",
-          width,
-          height,
-          entry,
-          observe,
-          unobserve,
-        };
-
-        if (breakpoints) {
-          e.currentBreakpoint = getCurrentBreakpoint(breakpoints, width);
-
-          if (e.currentBreakpoint !== prevBreakpointRef.current) {
-            if (onResizeRef.current) onResizeRef.current(e);
-            prevBreakpointRef.current = e.currentBreakpoint;
+
+          const next = {
+            currentBreakpoint: e.currentBreakpoint,
+            width,
+            height,
+            entry,
+          };
+
+          if (shouldUpdateRef.current && !shouldUpdateRef.current(next)) return;
+
+          if (
+            !shouldUpdateRef.current &&
+            breakpoints &&
+            updateOnBreakpointChange
+          ) {
+            rafId = requestAnimationFrame(() => {
```

Comment:
@cornelius-behrend Do we still need this?

Reply:
I tested it by commenting the two `requestAnimationFrame` lines out and it still works without error. So maybe that can be removed as well. I'm not sure about that `cancelAnimationFrame` - maybe this should also be added. I can make those changes in this PR if you agree.